src/pages/user_profile/AffiliationAndRoles.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {useEffect, useState} from 'react';
+import {Institution as InstitutionAPI} from '../../libs/ajax/Institution';
+import {Notifications} from '../../libs/utils';
+import {DuosUser, Institution} from '../../types/model';
+
+interface AffiliationAndRoleProps {
+  readonly user: DuosUser;
+}
+
+export default function AffiliationAndRole(props: AffiliationAndRoleProps) {
+
+  const {user} = props;
+  const [institution, setInstitution] = useState<Institution>();
+  const [roles, setRoles] = useState<string>('');
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        const allRoles = user?.roles?.map((role) => role.name).join(', ');
```

Comment:
Looking at the types, `user` and `user.role` are both non-optional. Are the types incorrect?
```suggestion
        const allRoles = user.roles.map((role) => role.name).join(', ');
```

Reply:
Correct, I'll clean that up